lib/buildingsync/makers/workflow_maker.rb

```diff
@@ -486,330 +771,79 @@ def gather_results(dir, baseline_only = false)
-                end_time.text = '2017-' + (month + 1).to_s + '-01T00:00:00'
-              else
-                end_time.text = '2018-01-01T00:00:00'
-              end
-              timeseries.add_element(end_time)
-              interval_frequency = REXML::Element.new("#{@ns}:IntervalFrequency")
-              interval_frequency.text = 'Month'
-              timeseries.add_element(interval_frequency)
-              interval_reading = REXML::Element.new("#{@ns}:IntervalReading")
-              the_key = "natural_gas_ip_#{month_lookup[month]}"
-              # puts "saving value: #{monthly_results[scenario_name][the_key.to_sym]}"
-              if !monthly_results[scenario_name][the_key.to_sym].nil?
-                interval_reading.text = monthly_results[scenario_name][the_key.to_sym] * 1000.0 # MMBtu to kBtu
-              end
-              timeseries.add_element(interval_reading)
-              resource_id = REXML::Element.new("#{@ns}:ResourceUseID")
-              resource_id.add_attribute('IDref', scenario_name_ns + '_NaturalGas')
-              timeseries.add_element(resource_id)
-              timeseriesdata.add_element(timeseries)
-            end
-            scenario.insert_after(res_uses, timeseriesdata)
-  
+            timeseries_data = get_timeseries_data_element(monthly_results,year_val, scenario_name)
+            scenario.insert_after(res_uses, timeseries_data)
+
             # all the totals
-            all_res_totals = REXML::Element.new("#{@ns}:AllResourceTotals")
-            all_res_total = REXML::Element.new("#{@ns}:AllResourceTotal")
-            end_use = REXML::Element.new("#{@ns}:EndUse")
-            end_use.text = 'All end uses'
-            site_energy_use = REXML::Element.new("#{@ns}:SiteEnergyUse")
-            site_energy_use.text = total_site_energy_kbtu.to_s
-            site_energy_use_intensity = REXML::Element.new("#{@ns}:SiteEnergyUseIntensity")
-            site_energy_use_intensity.text = total_site_eui_kbtu_ft2.to_s
-            source_energy_use = REXML::Element.new("#{@ns}:SourceEnergyUse")
-            source_energy_use.text = total_source_energy_kbtu.to_s
-            source_energy_use_intensity = REXML::Element.new("#{@ns}:SourceEnergyUseIntensity")
-            source_energy_use_intensity.text = total_source_eui_kbtu_ft2.to_s
-            all_res_total.add_element(end_use)
-            all_res_total.add_element(site_energy_use)
-            all_res_total.add_element(site_energy_use_intensity)
-            all_res_total.add_element(source_energy_use)
-            all_res_total.add_element(source_energy_use_intensity)
-            all_res_totals.add_element(all_res_total)
-            scenario.insert_after(timeseriesdata, all_res_totals)
-  
+            all_res_totals = get_all_resource_totals_element(total_site_energy_kbtu, total_site_eui_kbtu_ft2, total_source_energy_kbtu, total_source_eui_kbtu_ft2)
+            scenario.insert_after(timeseries_data, all_res_totals)
+
```

Comment:
Can we merge all the functions that are added data to the XML elements into one function?
Then we can add a test to check if this function is properly working